public/app/features/panel/panellinks/linkSuppliers.ts

```diff
@@ -14,27 +37,44 @@ export const getFieldLinksSupplier = (value: FieldDisplay): LinkModelSupplier<Fi
   }
   return {
     getLinks: (_scopedVars?: any) => {
-      const scopedVars: ScopedVars = {};
-      // TODO, add values to scopedVars and/or pass objects to event listeners
+      const scopedVars: DataLinkScopedVars = {};
+
       if (value.view) {
-        scopedVars[DataLinkBuiltInVars.seriesName] = {
+        const { dataFrame } = value.view;
+
+        scopedVars['__series'] = {
+          value: {
+            name: dataFrame.name,
+            labels: dataFrame.labels,
+            refId: dataFrame.refId,
+          },
           text: 'Series',
-          value: value.view.dataFrame.name,
         };
-        const field = value.column ? value.view.dataFrame.fields[value.column] : undefined;
+
+        const field = value.colIndex !== undefined ? dataFrame.fields[value.colIndex] : undefined;
+
         if (field) {
           console.log('Full Field Info:', field);
+          scopedVars['__field'] = {
+            value: {
+              name: field.name,
+            },
+            text: 'Field',
+          };
         }
-        if (value.row) {
-          const row = value.view.get(value.row);
-          console.log('ROW:', row);
-          const dataFrame = value.view.dataFrame;
 
+        if (value.rowIndex) {
           const { timeField } = getTimeField(dataFrame);
+
           if (timeField) {
-            scopedVars[DataLinkBuiltInVars.valueTime] = {
-              text: 'Value time',
-              value: timeField.values.get(value.row),
+            scopedVars['__value'] = {
+              value: {
+                raw: field.values.get(value.rowIndex),
+                numeric: value.display.numeric,
+                text: value.display.text,
+                time: timeField.values.get(value.rowIndex),
```

Comment:
This should not be nested under `if (timeField) {`

everything should exist except the time property when there is no time field